core/lib/basic_types/src/web3/mod.rs

```diff
@@ -72,7 +76,11 @@ impl<'a> Deserialize<'a> for Bytes {
     where
         D: Deserializer<'a>,
     {
-        deserializer.deserialize_identifier(BytesVisitor)
+        if deserializer.is_human_readable() {
+            deserializer.deserialize_identifier(BytesVisitor)
+        } else {
+            Vec::<u8>::deserialize(deserializer).map(Bytes)
+        }
```

Comment:
I recall there being a gotcha with this approach if the deserialized field is present in a `#[serde(flatten)]` group; I've encountered this in [my crate that implements similar functionality](https://github.com/slowli/hex-buffer-serde/blob/d203ab86d3e1772412728df19e3dca3d0eb6f447/src/var_len.rs#L305). Essentially, `serde` uses (or at least used in the past) a surrogate deserializer for such fields with `is_human_readable()` always set to true. This necessitates the original `BytesVisitor` supporting deserialization from bytes, even if it doesn't make logical sense. I guess in this case we can say that this field isn't supposed to be present in `#[serde(flatten)]` groups, but for completeness it could make sense to extend `BytesVisitor`.

Reply:
IIRC `bincode` doesn't support flattening anyways.

Reply:
anyways... added some `BytesVisitor` code ... plz re-review